src/manager.rs

```diff
@@ -0,0 +1,132 @@
+use std::io::prelude::*;
+use std::fs::File;
+use std::fs;
+use std::io::{ Error, ErrorKind };
+use compile::ExecutableDevEnv;
+use std::collections::HashMap;
+use parse::Parser;
+use std::marker::PhantomData;
+use run::Execution;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use run::ExecutionEvent;
+use ast::{ Script, UncheckedCtx };
+use std::thread;
+
+struct ScriptManager<Env> where Env: ExecutableDevEnv + Sync + 'static {
+    path: String,
+    phantom: PhantomData<Env>,
+    api: Env::API,
```

Comment:
Actually, you just need an instance of `Env`. It will provide you with the instance of `API`.